### DISCUSSION TITLE
New to Homebrew and far from expert with Linux?

### DISCUSSION THREAD

**Discussion by ejwjohn** [Homebrew/discussions#756] [Getting started]
Hello,

I operate a MAC OSx on 11.2 (Big Sur)

I was in the process of installing Homebrew, and I believe it had completed the Xcode  additions.

Then I had an Ethernet glitch which stopped the transmission which would not recover, my only option was to terminate Terminal and start over.

I then started getting errors that were 

..HEAD is now at 455af8523 Merge pull request #10547 from Rylan12/prepare-bottle-syntax-deprecations
fatal: Could not resolve HEAD to a revision
==> Installation successful!

I looked on a couple of other web site forums and found similar problems reported but without any fix, so thought I better get in touch with the developer Forum page..

What I have done to date is this 
1. On one of the other forums it suggested running the following command
- /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
- but it had no impact on the issue.
2. I then deleted the files related to Homebrew installation in the following
3.- /usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
- also not impact on running the install command.

Apologies for the limited information I can provide...

Any chance of a resolution, please?

Thanks

JohnW

**Comment by SMillerDev** [comment]
- Do you have an Apple Silicon based mac?
- Did any of the commands output anything?
- When are you getting these errors you mention?

**Reply by ejwjohn** [reply]
Hello,

No this is an Intel MAC.
The current output is as follows:

Mac-mini:~ JohnWilliams$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew

Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /usr/sbin/chown -R JohnWilliams:admin /usr/local/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at 455af8523 Merge pull request #10547 from Rylan12/prepare-bottle-syntax-deprecations
fatal: Could not resolve HEAD to a revision
==> Installation successful!

==> Homebrew has enabled anonymous aggregate formulae and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics data has been sent yet (or will be during this `install` run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

==> Next steps:
- Run `brew help` to get started
- Further documentation: 
    https://docs.brew.sh
Mac-mini:~ JohnWilliams$ 

Q3 3 is answered as above.

Thank You

Johnw

**Reply by SMillerDev** [reply]
> I then started getting errors that were

> ..HEAD is now at 455af8523 Merge pull request #10547 from Rylan12/prepare-bottle-syntax-deprecations
fatal: Could not resolve HEAD to a revision
==> Installation successful!

Errors rarely pop up without user input. What do you do when these happen? Is it just during the install? Is it in `brew help`? Is it when you open your terminal?

**Reply by ejwjohn** [reply]
Hello,

No, you can see the Fatal  Could not resolve Head, not sure why it goes onto say installation is successful.....

I only input the one command line and then I get the output as attached above....

Also, the following files appear to be installed but are they correct, I would not know.

/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew

Thanks for your assistance

JohnW

**Reply by ejwjohn** [reply]
OK,

Update,

When I noted the error and the word FATAL I assumed this was really bad news.... so did not progress any further.

The purpose of loading Homebrew as to load VIciaul Studio and Python....

So I tested if Homebrew was loaded correctly using the Comand -V Brew and which brew commands and they seemed to show everything was OK.

ANd starting VS Code seems to be OK.

So do I need to be concerned about the Fatal Error ???

Thanks

JohnW

**Reply by SMillerDev** [reply]
> The purpose of loading Homebrew as to load Visual Studio and Python....

Neither of these require homebrew to be installed. There is a `brew doctor` command you can run to make homebrew do a self-test, but it doesn't seem like you need to.

**Reply by ejwjohn** [reply]
I googled and the general consensus was that to install VSC and Python using Homebrew

JohnW
